src/apps/api/views/profiles.py

```diff
@@ -14,3 +19,28 @@ class GetMyProfile(RetrieveAPIView, GenericAPIView):
 
     def get_object(self):
         return self.request.user
+
+
+@login_required
+def user_lookup(request):
+    search = request.GET.get('q', '')
+    filters = Q()
+    is_admin = request.user.is_superuser or request.user.is_staff
+
+    if search:
+        filters |= Q(username__icontains=search)
+        filters |= Q(email__icontains=search) if is_admin else Q(email__iexact=search)
+
+    users = User.objects.filter(filters)[:5]
+
+    # Helper to print username with email for admins
+    def _get_data(user):
+        return {
+            "id": user.id,
+            "username": user.username,
+            "name": f"{user.name} ({user.email})" if is_admin else user.username,
+        }
+
+    return HttpResponse(
+        json.dumps({"results": [_get_data(u) for u in users]}),
+    )
```

Comment:
Slightly refactored this from v1.5, but largely kept things the same. Based on Isabelle's comment about this feature being slow on 1.5, is that a function of the logic of this feature, or that page in general? Or rather, is there a better way I should be adding this functionality that won't cause speed problems down the road?

Reply:
That page in general. It is a bunch of inline formsets that generate hundreds of queries, on that page.